src/main/java/org/zalando/jackson/datatype/money/MoneyModule.java

```diff
@@ -64,37 +61,28 @@ public Version version() {
     @Override
     public void setupModule(final SetupContext context) {
         final SimpleSerializers serializers = new SimpleSerializers();
-
-        serializers.addSerializer(CurrencyUnit.class, new CurrencyUnitSerializer());
-        serializers.addSerializer(NumberValue.class, numberValueSerializer);
-        serializers.addSerializer(MonetaryAmount.class, new MonetaryAmountSerializer(formatFactory, names));
-
+        serializers.addSerializer(MonetaryAmount.class, new MonetaryAmountSerializer(names, writer, formatFactory));
         context.addSerializers(serializers);
 
         final SimpleDeserializers deserializers = new SimpleDeserializers();
-
-        deserializers.addDeserializer(CurrencyUnit.class, new CurrencyUnitDeserializer());
-        deserializers.addDeserializer(NumberValue.class, new DecimalNumberValueDeserializer());
         deserializers.addDeserializer(MonetaryAmount.class, new MonetaryAmountDeserializer<>(amountFactory, names));
-
         // for reading into concrete implementation types
         deserializers.addDeserializer(Money.class, new MonetaryAmountDeserializer<>(moneyFactory, names));
         deserializers.addDeserializer(FastMoney.class, new MonetaryAmountDeserializer<>(fastMoneyFactory, names));
         deserializers.addDeserializer(RoundedMoney.class, new MonetaryAmountDeserializer<>(roundedMoneyFactory, names));
-
         context.addDeserializers(deserializers);
     }
 
     public MoneyModule withDecimalNumbers() {
-        return withNumbers(new DecimalNumberValueSerializer());
+        return withNumbers(new DecimalAmountWriter());
     }
 
     public MoneyModule withQuotedDecimalNumbers() {
-        return withNumbers(new QuotedDecimalNumberValueSerializer());
+        return withNumbers(new QuotedDecimalAmountWriter());
     }
 
-    public MoneyModule withNumbers(final JsonSerializer<? super NumberValue> numberValueSerializer) {
-        return new MoneyModule(numberValueSerializer, names, formatFactory, amountFactory,
+    private MoneyModule withNumbers(final AmountWriter writer) {
```

Comment:
Making this private will allow us to implement features like this easier in the future. I'd also argue that registering a serializer for the amount itself is too narrow/restricting anyway to be useful.

Reply:
Agreed 😺